test/LockedNORI.test.ts

```diff
@@ -0,0 +1,1166 @@
+        expect(
+          await lNori.hasRole(await lNori[role](), hre.namedAccounts.admin)
+        ).to.be.true;
+        expect(await lNori.getRoleAdmin(await lNori[role]())).to.eq(
+          await lNori.DEFAULT_ADMIN_ROLE()
+        );
+        expect(await lNori.getRoleMemberCount(await lNori[role]())).to.eq(1);
+      });
+    }
+  });
+});
+
+describe('role access', () => {
+  // it.todo('only the token granter role can deposit tokens')
+  describe('roles', () => {
+    describe('TOKEN_GRANTER_ROLE', () => {
+      for (const { role, accountWithRole, accountWithoutRole } of [
+        {
+          role: 'TOKEN_GRANTER_ROLE',
+          accountWithRole: 'admin',
+          accountWithoutRole: 'investor1',
+        } as const,
+      ]) {
+        it(`accounts with the role "${role}" can use "batchRevokeUnvestedTokenAmounts" whilst accounts without the role "${role}" cannot`, async () => {
+          const { lNori, grantAmount, grant, hre } = await setupWithGrant(
+            (parameters) => employeeParameters(parameters)
+          );
+          const { namedAccounts, namedSigners } = hre;
+          const roleId = await lNori[role]();
+          expect(await lNori.hasRole(roleId, namedAccounts[accountWithRole])).to
+            .be.true;
+          expect(
+            await lNori
+              .connect(namedSigners[accountWithRole])
+              .batchRevokeUnvestedTokenAmounts(
+                [grant.recipient],
+                [namedAccounts.admin],
+                [grant.startTime + DELTA],
+                [0]
+              )
+          )
+            .to.emit(lNori, 'UnvestedTokensRevoked')
+            .withArgs(
+              grant.startTime + DELTA,
+              namedAccounts.employee,
+              grantAmount
+            );
+          // todo test balance of admin is now the revoked token balance
+          await expect(
+            lNori
+              .connect(namedSigners[accountWithoutRole])
+              .batchRevokeUnvestedTokenAmounts(
+                [grant.recipient],
+                [namedAccounts.admin],
+                [grant.startTime + DELTA],
+                [0]
+              )
+          ).to.be.revertedWith(
+            `AccessControl: account ${namedAccounts[
+              accountWithoutRole
+            ].toLowerCase()} is missing role ${roleId}`
+          );
+        });
+        it(`accounts with the role "${role}" can use "createGrant" whilst accounts without the role "${role}" cannot`, async () => {
+          const { lNori, hre } = await setupTest();
+          const { namedAccounts, namedSigners } = hre;
+          const roleId = await lNori[role]();
+          expect(await lNori.hasRole(roleId, namedAccounts[accountWithoutRole]))
+            .to.be.false;
+          const { grant, grantAmount } = employeeParameters({
+            hre,
+            startTime: await getLatestBlockTime({ hre }),
+          });
+          expect(
+            await lNori
```

Comment:
[await order is important for this](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events)
```suggestion
          await expect(
            lNori
```